test/lib/checkevents.go

```diff
@@ -0,0 +1,243 @@
+	egi, err := newEventGetter(podName, client, logf)
+	if err != nil {
+		return nil, err
+	}
+	ei := newTestableEventInfoStore(egi, -1, -1)
+	return ei, nil
+}
+
+// Starts the single threaded background goroutine used to update local state
+// from the remote REST API.
+func (ei *EventInfoStore) start() {
+	ei.closeCh = make(chan struct{})
+	ei.doRefresh = make(chan chan error)
+	go func() {
+		for {
+			select {
+			case <-ei.closeCh:
+				ei.getter.cleanup()
+				return
+			case replyCh := <-ei.doRefresh:
+				replyCh <- ei.doRetrieveData()
+			}
+		}
+	}()
+}
+
+// The data update thread used by the single threaded background goroutine
+// for updating data from the REST api.
+func (ei *EventInfoStore) doRetrieveData() error {
+	min, max, err := ei.getter.getMinMax()
+	if err != nil {
+		return fmt.Errorf("error getting MinMax %v", err)
+	}
+	ei.lock.Lock()
+	curMin := ei.firstID
+	curMax := curMin + len(ei.allEvents) - 1
+	ei.lock.Unlock()
+	if min == max+1 {
+		// Nothing to read or trim
+		return nil
+	} else {
+		if min > curMax+1 {
+			return fmt.Errorf("mismatched stored max/available min: %d, %d", curMax, min)
+		}
+		min = curMax + 1
+		// We may have data to read, definitely have data to trim.
+	}
+	var newEvents []EventInfo
+	for i := min; i <= max; i++ {
+		e, err := ei.getter.getEntry(i)
+		if err != nil {
+			return fmt.Errorf("error calling getEntry of %d %v", i, err)
+		}
+
+		newEvents = append(newEvents, e)
+	}
+	ei.lock.Lock()
+	ei.allEvents = append(ei.allEvents, newEvents...)
+	ei.lock.Unlock()
+	err = ei.getter.trimThrough(max)
+	return err
+
+}
+
+// Clean up any background resources used by the store.  Must be called exactly once after
+// the last use.
+func (ei *EventInfoStore) Cleanup() error {
+	close(ei.closeCh)
+	return nil
+}
+
+// Called internally by functions wanting the current list of all
+// known events.  This calls for an update from the REST server and
+// returns the summary of all locally and remotely known events.
+// Returns an error in case of a connection or protocol error.
+func (ei *EventInfoStore) refreshData() ([]EventInfo, error) {
+	var allEvents []EventInfo
+	replyCh := make(chan error)
+	ei.doRefresh <- replyCh
+	err := <-replyCh
+	if err != nil {
+		return nil, err
+	}
+	ei.lock.Lock()
+	allEvents = append(allEvents, ei.allEvents...)
+	ei.lock.Unlock()
+	return allEvents, nil
+}
+
+// Find all events received by the recordevents pod that match the provided function,
+// returning all matching events as well as a SearchedInfo structure including the
+// last 5 events seen and the total events matched.  This SearchedInfo structure
+// is primarily to ease debugging in failure printouts.  The provided function is
+// guaranteed to be called exactly once on each EventInfo from the pod.
+func (ei *EventInfoStore) Find(f EventInfoMatchFunc) ([]EventInfo, SearchedInfo, error) {
+	const maxLastEvents = 5
```

Comment:
nit: maybe this max can be part of the EventInfoStore? If not specified, then default to 5.
Again, no strong opinion

Reply:
same comment as the timeout